### ISSUE TITLE
Incorrect signature local crash with accent on a eas configured application

### ISSUE THREAD

**Issue by soullivaneuh** [expo/expo#19431]
### Summary

Before reading further, grab a coffee and take a seat. It will certainly brain your mind. :upside_down_face: 

I am developing a French application, and in France, we like special chars like accents.

EAS however, does not seem to like them. Configuring it on an expo project on providing sentence with special chars on some `infoPlist` properties make the app loaded from `npx expo start` ending to the crash with the following:

```
Uncaught Error: java.io.IOException: Manifest download was successful, but signature was incorrect
```

1. Removing the accent "solves" the issue
2. Removing the EAS configuration (`runtimeVersion`, `projectId`) also "solves" the issue (you said magic?)

Here is the possible workarounds: 

- Remove the special chars from your sentences
- Use [localization files](https://docs.expo.dev/distribution/app-stores/#localizing-your-ios-app) to store the sentences

I chose the first one for my case. So I am not stuck anymore, however, I am very curious to see the explanation of that weird behavior! :D

### What platform(s) does this occur on?

Android, iOS

### SDK Version

46

### Environment

  expo-env-info 1.0.4 environment info:
    System:
      OS: Linux 5.15 Manjaro Linux
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 16.15.1 - ~/.nvm/versions/node/v16.15.1/bin/node
      Yarn: 1.22.19 - /usr/bin/yarn
      npm: 8.19.2 - ~/.nvm/versions/node/v16.15.1/bin/npm
    IDEs:
      Android Studio: AI-213.7172.25.2113.9014738
    npmPackages:
      expo: ~46.0.13 => 46.0.15 
      react: 18.0.0 => 18.0.0 
      react-dom: 18.0.0 => 18.0.0 
      react-native: 0.69.6 => 0.69.6 
      react-native-web: ~0.18.7 => 0.18.9 
    npmGlobalPackages:
      eas-cli: 2.2.1
      expo-cli: 6.0.6
    Expo Workflow: managed

### Minimal reproducible example

Create a new blank project using `npx create-expo-app my-app --template` (I used Typescript configuration).

Put some accents on your `app.config.ts`, it should looks like this:

```typescript
import {
  ExpoConfig,
} from '@expo/config';

const nsLocationDescription = "ééé";

export default (): ExpoConfig => ({
  name: 'my-app',
  slug: 'my-app',
  ios: {
    infoPlist: {
      NSLocationWhenInUseUsageDescription: nsLocationDescription,
      NSLocationAlwaysAndWhenInUseUsageDescription: nsLocationDescription,
      NSLocationAlwaysUsageDescription: nsLocationDescription,
    },
  },
});
```

If you start the application at this point, it will work properly.

Then, run `eas init` and `eas update:configure`, it should ends with the following changes on the config:

```diff
diff --git a/app.config.ts b/app.config.ts
index 509b47a..10226ce 100644
--- a/app.config.ts
+++ b/app.config.ts
@@ -7,6 +7,12 @@ const nsLocationDescription = "ééé";
 export default (): ExpoConfig => ({
   name: 'my-app',
   slug: 'my-app',
+  runtimeVersion: {
+    policy: 'sdkVersion',
+  },
+  updates: {
+    url: 'https://u.expo.dev/XXX',
+  },
   ios: {
     infoPlist: {
       NSLocationWhenInUseUsageDescription: nsLocationDescription,
@@ -14,4 +20,9 @@ export default (): ExpoConfig => ({
       NSLocationAlwaysUsageDescription: nsLocationDescription,
     },
   },
+  extra: {
+    eas: {
+      projectId: 'XXX',
+    },
+  },
 });
```

Finally, run `npm start` and try to open the project with Expo Go.

**Comment by douglowder** [comment]
Thanks for this report. Indeed mind blowing, and we will take a look 😀

**Comment by KurtMar** [comment]
I stumbled upon this bug myself while upgrading an app to EAS. ```npx expo start --ios``` was giving me the same ```Manifest download was successful, but signature was incorrect``` error message in Expo Go on iOS Simulator, so I happily proceeded testing locally with the legacy global client. After submitting my EAS build to Apple and running it on my device, it would simply crash after the splash screen.

In my case the issue is umlauts in the app name. Removing the single character I have results in Expo Go loading the app correctly. This means that this should be easily reproduced by simply changing the name in app.json of any working app to e.g.:
```
{
  "expo": {
    "name": "öäå"
    ...
}
```

This issue is likely a showstopper for anyone upgrading to EAS with an app name containing _special_ characters.

**Comment by douglowder** [comment]
Since `Info.plist` is XML markup, many special characters have to be escaped. I tried the following and it got past the error you are seeing:

```ts
import {
  ExpoConfig,
} from '@expo/config';

const nsLocationDescription = "ééé";

export default (): ExpoConfig => ({
  name: 'my-app',
  slug: 'my-app',
  ios: {
    infoPlist: {
      NSLocationWhenInUseUsageDescription: nsLocationDescription,
      NSLocationAlwaysAndWhenInUseUsageDescription: nsLocationDescription,
      NSLocationAlwaysUsageDescription: nsLocationDescription,
    },
  },
});
```

**Comment by douglowder** [comment]
I'll have a look now at what should be done when the app name has special characters.

**Comment by fossecode** [comment]
I am experiencing the same issue when I have special characters in `NSLocationWhenInUseUsageDescription`, `NSLocationAlwaysAndWhenInUseUsageDescription` and `NSLocationAlwaysUsageDescription`.

**Comment by IlyaSemenov** [comment]
As mentioned above, this is not only `name` that causes issues, I'm having the same problem with:

```ts
{
  ios: {
    infoPlist: {
      NSLocationWhenInUseUsageDescription:
        "Приложение использует GPS для отслеживания положения грузовиков.",
    }
  }
}
```

**Comment by jmgarciasanti** [comment]
Same here with SDK 46. Accents in "displayName" or in "description" in app.json throws the "...signature was incorrect..." error. Both in Expo Go for Android and iOS

**Comment by chris-metz** [comment]
Confirmed, having the same issue with German umlauts in NSCalendarsUsageDescription. Works after removing them (expo 46).

**Comment by wschurman** [comment]
Looking. I was able to repro using the app.json. Will report back when I know more about a possible fix.

**Comment by vinaciotm** [comment]
Same problem pt-BR language in `app.json ` with keys `description` and  `infoPlist` in `ios`

**Comment by akifmertturk** [comment]
Hi everyone, @wschurman, @douglowder 

I got the same error and figured it out.

Whenever I run the `eas update..` command, _app.json_ file is updated, for example, new lines are added for updated info, url, etc.
And the problem occurs here, because of these lines, the Expo app could not run on the emulator. Whenever I deleted those newly added lines after `eas update`, the app works in a regular way without any warnings or errors.
